waku/v2/utils/keyfile.nim

```diff
@@ -0,0 +1,532 @@
+# This implementation is originally taken from nim-eth keyfile module https://github.com/status-im/nim-eth/blob/master/eth/keyfile and adapted to 
```

Comment:
I guess there are some parts from the `keyfile` module that are imported here without any change, right? if yes, what do you think about importing such untouched parts directly from the `nim-eth/keyfile` module and just implementing the differences (which enable arbitrary data size encryption and other features) in the current module?  some immediate benefits are -it makes the review of new edits easier, and -it is less error-prone 
wdyt?